detect_secrets/core/audit.py

```diff
@@ -332,7 +339,10 @@ def print_audit_results(baseline_filename):
 def _get_baseline_from_file(filename):  # pragma: no cover
     try:
         with open(filename) as f:
-            return json.loads(f.read())
+            baseline = json.loads(f.read())
+            if 'custom_plugin_paths' in baseline:
+                baseline['custom_plugin_paths'] = tuple(baseline['custom_plugin_paths'])
```

Comment:
```diff
Kevins-MacBook-Pro:detect-secrets kevin$ gds
diff --git a/detect_secrets/core/audit.py b/detect_secrets/core/audit.py
index 92d418b..d79a206 100644
--- a/detect_secrets/core/audit.py
+++ b/detect_secrets/core/audit.py
@@ -341,6 +341,8 @@ def _get_baseline_from_file(filename):  # pragma: no cover
         with open(filename) as f:
             baseline = json.loads(f.read())
             if 'custom_plugin_paths' in baseline:
+                # We turn custom_plugins_paths into a tuple so that we can
+                # @lru_cache all the functions that take it as an argument.
                 baseline['custom_plugin_paths'] = tuple(baseline['custom_plugin_paths'])
             return baseline
     except (IOError, json.decoder.JSONDecodeError):
diff --git a/detect_secrets/core/usage.py b/detect_secrets/core/usage.py
index eacbf71..dc54445 100644
--- a/detect_secrets/core/usage.py
+++ b/detect_secrets/core/usage.py
@@ -50,6 +50,10 @@ class TupleAction(argparse.Action):
 
 
 def add_custom_plugins_argument(parser):
+    """
+    We turn custom_plugins_paths into a tuple so that we can
+    @lru_cache all the functions that take it as an argument.
+    """
     parser.add_argument(
         '--custom-plugins',
         action=TupleAction,
```
👍